SonarQube.Common/FileConstants.cs

```diff
@@ -47,21 +47,47 @@ public static class FileConstants
         /// </summary>
         public const string IntegrationTargetsName = "SonarQube.Integration.targets";
 
+        private static bool IsRunningOnWindows()
+        {
+            return Environment.OSVersion.Platform == PlatformID.Win32NT;
+        }
+
         /// <summary>
         /// Path to the user specific ImportBefore folders
         /// </summary>
         public static IReadOnlyList<string> ImportBeforeDestinationDirectoryPaths
         {
             get
             {
-                string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
-                return new string[]
-                {
-                    Path.Combine(appData, "Microsoft", "MSBuild", "15.0", "Microsoft.Common.targets", "ImportBefore"),
-                    Path.Combine(appData, "Microsoft", "MSBuild", "14.0", "Microsoft.Common.targets", "ImportBefore"),
-                    Path.Combine(appData, "Microsoft", "MSBuild", "12.0", "Microsoft.Common.targets", "ImportBefore")
-                };
+                return IsRunningOnWindows() ? GetWindowsFolders() : GetNonWindowsFolders();
             }
         }
+
+        private static IReadOnlyList<string> GetWindowsFolders()
+        {
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            return new string[]
+            {
+                Path.Combine(appData, "Microsoft", "MSBuild", "15.0", "Microsoft.Common.targets", "ImportBefore"),
+                Path.Combine(appData, "Microsoft", "MSBuild", "14.0", "Microsoft.Common.targets", "ImportBefore")
+            };
+        }
+
+        private static IReadOnlyList<string> GetNonWindowsFolders()
+        {
+            string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+
+            return new string[]
+            {
+                // for msbuild
+                Path.Combine(appData, "Microsoft", "MSBuild", "15.0", "Microsoft.Common.targets", "ImportBefore"),
```

Comment:
Seems to be duplicate with GetWindowsFolders. Couldn't you extract common part of the two methods?